app/assets/javascripts/initializers/initializeBaseTracking.js.erb

```diff
@@ -116,6 +116,29 @@ function trackCustomImpressions() {
         successLinks[i].addEventListener('click', function() { trackHtmlVariantSuccess(dataBody, csrfToken) });
       }
     }
+
+    // page view
+    if (ArticleElement && tokenMeta && !isBot) {
+      var randomNumber = Math.floor(Math.random() * 10); // 1 in 10; Only track 1 in 10 impressions
```

Comment:
Why don't you want to track all views?

Reply:
I suppose that it's done for optimization, to make fewer requests and db writes. For unauthorized users, only one of each 10 requests will be saved, but the count will be 10 (https://github.com/thepracticaldev/dev.to/pull/1985/files#diff-a71fed5e724b3cd9cce62138dfb4d631R10), so the number of views will remain approximately the same as if every request was tracked.

Reply:
Yup, that’s the logic. We can adjust things later, but this should be a bit less resource costly and give us good enough insights.